otherlibs/unix/write_win32.c

```diff
@@ -145,3 +145,30 @@ CAMLprim value caml_unix_single_write(value fd, value buf, value vofs,
   }
   CAMLreturn(Val_long(written));
 }
+
+intnat caml_unix_fast_single_write(value fd, value buf, intnat ofs, intnat len)
+{
+  intnat ret = -1;
+  if (len > 0) {
+    if (Descr_kind_val(fd) == KIND_SOCKET) {
+      SOCKET s = Socket_val(fd);
+      ret = send(s, &Byte(buf, ofs), len, 0);
+      if (ret == SOCKET_ERROR) ret = -1;
+    } else {
+      HANDLE h = Handle_val(fd);
+      DWORD writtenwords;
+      if (WriteFile(h, &Byte(buf, ofs), ofs, &writtenwords, NULL))
```

Comment:
```suggestion
      if (WriteFile(h, &Byte(buf, ofs), len, &writtenwords, NULL))
```
I believe the third argument is the length to write (there is no offset argument to this function).